SourceX/miniwin/misc_msg.cpp

```diff
@@ -406,13 +406,9 @@ WINBOOL PeekMessageA(LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilter
 			break;
 		case GameActionType::CAST_SPELL:
 			if (pcurs >= CURSOR_FIRSTITEM && invflag)
-				// Drop item so that it does not get destroyed.
 				DropItemBeforeTrig();
-			if (!invflag && !talkflag)
-				// Cast the spell.
-				RightMouseDown();
-			// Close active menu if any / stop walking.
-			PressEscKey();
+			else

```

Comment:
Should this be in the `else` branch? Previously the spell was cast even if the inventory was open

Reply:
It feelt better to have only one function for the key at one time, this way you can drop items with out also casting a random firewall :)